components/browser/session/src/main/java/mozilla/components/browser/session/SelectionAwareSessionObserver.kt

```diff
@@ -47,7 +47,7 @@ abstract class SelectionAwareSessionObserver(
     /**
      * Stops the observer.
      */
-    fun stop() {
+    open fun stop() {
```

Comment:
We had to add this one to be able to override the method.

Reply:
Instead of extending `SelectionAwareSessionObserver` you could also create a local instance inside `DownloadsFeature`. May make it easier to understand what's going on too. :)

Reply:
But it will also over-complicate things a bit.

Because then I will have to implement `Session.Observer` for getting notify on `onDownload` and I will need the `activeSession` and `SelectionAwareSessionObserver` just expose the `activeSession` for their children 😿  .

https://github.com/mozilla-mobile/android-components/blob/996c9c493beb21a182b08680889fa5df6e4b2a36/components/browser/session/src/main/java/mozilla/components/browser/session/SelectionAwareSessionObserver.kt#L23